From 438f07d85abf6a351c936a1578a31f1073f4abc7 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Tue, 25 Nov 2014 19:48:28 -0600
Subject: [PATCH] Declare new thrown exceptions in generateKeys() (bc1.47)

Thankfully, generateKeys() is private, so declaring the exceptions
doesn't break anything.  generate() is the only public method that calls
it, already can throw NoSuchAlgorithmException, and I catch the
other two and rethrow an Error.
---
 src/org/glite/security/util/proxy/ProxyCertificateGenerator.java | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/org/glite/security/util/proxy/ProxyCertificateGenerator.java b/src/org/glite/security/util/proxy/ProxyCertificateGenerator.java
index e4cf2fe..3441844 100644
--- a/src/org/glite/security/util/proxy/ProxyCertificateGenerator.java
+++ b/src/org/glite/security/util/proxy/ProxyCertificateGenerator.java
@@ -19,6 +19,7 @@ package org.glite.security.util.proxy;
 import java.io.IOException;
 import java.io.StringWriter;
 import java.math.BigInteger;
+import java.security.InvalidAlgorithmParameterException;
 import java.security.InvalidKeyException;
 import java.security.KeyPair;
 import java.security.KeyPairGenerator;
@@ -250,7 +251,11 @@ public class ProxyCertificateGenerator {
                         "Only private key of the proxy is set. As it is, also public key has to be set.");
             }
 
-            generateKeys();
+            try {
+                generateKeys();
+            } catch (InvalidAlgorithmParameterException|NoSuchProviderException e) {
+                throw new Error("(should not happen) from generateKeys(): ", e);
+            }
         }
 
         // default to RFC proxy id all else fails.
@@ -283,7 +288,7 @@ public class ProxyCertificateGenerator {
     /**
      * Generates the private and public keys if they are not given.
      */
-    private void generateKeys() {
+    private void generateKeys() throws InvalidAlgorithmParameterException, NoSuchAlgorithmException, NoSuchProviderException {
         KeyPairGenerator keyPairGen = KeyPairGenerator.getInstance("RSA", "BC");
         keyPairGen.initialize(
             new RSAKeyGenParameterSpec(m_keyLength,
-- 
2.0.1

